src/notifications.js

```diff
@@ -12,7 +12,7 @@ class Notifications extends React.Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    const {notifications} = nextProps;
+    const notifications = nextProps.notifications.toJS();
```

Comment:
You are definitely wrong, @alameya 
"Map" creates new object each time, so you force component to rerender each time you call reducer. This is drastically decrease perfomance.